release-notes.txt

```diff
@@ -1,3 +1,521 @@
+real-time capabilities, small memory footprint, modularity, and uniform API
+access, independent of the underlying hardware (with partial POSIX compliance).
+
+RIOT is developed by an international open-source community which is
+independent of specific vendors (e.g. similarly to the Linux community) and is
+licensed with a non-viral copyleft license (LGPLv2.1), which allows indirect
+business models around the free open-source software platform provided by RIOT.
+
+
+About this release:
+===================
+
+The 2019.10 release includes:
+
+ - initial support for SUIT firmware updates
+ - USB CDC-ACM serial communication
+ - complete rewrite of TI CC110x radio driver
+ - initial support for IPv6 fragmentation
+ - DTLS support in the sock networking stack
+ - complete blockwise messaging for gcoap and nanocoap
+ - as always, bug fixes and documentation updates
+
+About 460 pull requests, composed of 950 commits, have been merged since the
+last release, and about 70 issues have been solved. 57 people contributed with
+code in 105 days. Approximately 2000 files have been touched with 129000
+insertions and 25000 deletions.
+
+
+Notations used below:
+=====================
+    + means new feature/item
+    * means modified feature/item
+    - means removed feature/item
+
+New features and changes
+========================
+
+System libraries
+----------------
+    + sys/arduino: Added millis()
+    * sys/arduino: make ADC feature optional
+    + sys/fmt: Added submodule fmt_table for printing tables
+    + sys/log: add module for colorized logging
+    + sys/riotboot: add initial image digest verification
+    + sys/shell: add heap command
+    + sys/shell: cancel current line on CTRL-C.
+    * shell/sc_gnrc_ipv6_nib: list and edit authoritative border router (ABR) list
+    + sys/suit: initial support for SUIT firmware updates
+    * sys: add schedstatistics module to remove dependency on xtimer
+    * can: use memarray for pkt and router allocation
+    + sys: single interrupt handler thread for interrupts in modules with blocking
+      functions (but likely to be removed -- see Deprecations section below)
+
+Networking
+----------
+    + sys: Add (CoAP) Link Format module
+    + sys: add credman (D)TLS credential management module
+    + sys: new sock submodule for DTLS
+    + gnrc_ipv6_ext_frag: Initial import of IPv6 fragmentation and reassembly
+    + gnrc_netif: allow for wait of minimum time between sends
+    * gnrc_netif: add netif setter (part of refactoring to a pointer-based
+      approach)
+    * gnrc_sixlowpan_frag_vrb: add gnrc_sixlowpan_frag_vrb_from_route()
+    + sixlowpan_sfr: initial import Selective Fragment Recovery definitions
+    + gcoap: add Blockwise support
+    * nanocoap: complete Blockwise support
+    + nanocoap: add coap_opt_get_next() to iterate over options
+    + nanocoap: add coap_opt_get_opaque() to retrieve option bytes
+    * tcp: several small improvements
```

Comment:
These are mostly fixes as far as I can see:

```bash
git log --no-decorate --no-merges --oneline 2019.10-devel..2019.10-branch -- sys/include/net/tcp.h sys/include/net/gnrc/tcp sys/net/transport_layer/tcp/ sys/net/gnrc/transport_layer/tcp/
```

> 4fbc0252e1 gnrc_tcp: do not reopen recv window on closed connection
> 66e5f7230f gnrc_tcp: check if option has valid length set
> 07210ee571 gnrc_tcp: fix debug message
> fdda22c200 gnrc_tcp: allow unknown options
> e5503d62bf gnrc_tcp: check if option length field is present before accessing it
> 463c1212c2 gnrc_tcp: Always ensure that option->length doesn't exceed opt_left
> 9764697a3b gnrc_tcp: abort on unknown option
> 351220e884 gnrc_tcp - Release pakets on failed gnrc_netapi_send()
> 9cf0c3bc70 sys/net: fix typos in docu
> 7f3415e91d gnrc_tcp: Fix invocation of gnrc_pktbuf_add
> c9358b9217 gnrc_tcp: fix integer underflow in option parser
> 0fff1b35ec gnrc_tcp: add minimum length check to _receive
> c6df2064af gnrc_tcp: Fix memory leak
> 9e91d21625 net/gnrc/tcp: fix uninitialized memory access
> e352b54a9b net/gnrc/tcp: remove duplicate net_gnrc_tcp group definitions

If you still want to keep it

```suggestion
    * gnrc_tcp: several small improvements
```

Reply:
Fixed in d8aafc0